nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server.hpp

```diff
@@ -55,6 +55,33 @@ class BtActionServer
     OnPreemptCallback on_preempt_callback,
     OnCompletionCallback on_completion_callback);
 
+  /**
```

Comment:
After reading this, I think I understand the intention in the PR. You're seeing that we have multiple BT action servers in the 2 navigators and they share a name so you want to have them be separated. Got it. 

Do you care the names they receive? The client node is sent to the blackboard for use in the different BT node options, and they wouldn't be namespaced under the node at all. As such, I think it would be safe to say "no" we don't care about the specific string name of the node.

If that's the case, a much simpler solution to this issue is to do what TF does https://github.com/ros2/geometry2/blob/ros2/tf2_ros/src/transform_listener.cpp#L48 and just apply a UUID or hex after the topic context name to randomize the name to prevent collisions. That would involve no ABI/API breaking changes and we could safely merge that into both Galactic and `main` this week easily .

Reply:
> apply a UUID or hex after the topic context name to randomize the name to prevent collisions.

Although, nondeterministic node names are a pain for security, as there's no good way to fix the runtime DDS security permissions needed before at design time, while still being node specific.

Reply:
What's the plan for TF on that front (or is there)? they could do something like `tf_impl_listener_<name of node given it it that should be unique>`